### PR TITLE
User Directory + Search User for Team

### PR DESCRIPTION
**Pull request by akbarsahata**
## Changes

- [ADD] endpoints user directory
- [ADD] endpoints search student for team
- [MODIFY] refactored common query/params interface
- [MODIFY] activate logger

## Reviewer
@yofriadi @ibnuimam @azhariemuhammad 

## Links
- [Postman JSON file](https://api.jsonbin.io/b/5ffaee6255b359028dbd311b)

### CODE REVIEW

File: src/api/controllers/user.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      '',
+      req.query.limit,
+      req.query.skip,
+    );
+
+    return userTeachers;
+  }
+
+  @GET({
+    url: '/my/teachers',
```
Comment (azhariemuhammad) [inline src/api/controllers/user.ts:38]: @akbarsahata  yang gw pake untuk idea ini ya?

Comment (akbarsahata) [reply src/api/controllers/user.ts:38]: Iya ini buat cari teachernya rie

Comment (azhariemuhammad) [reply src/api/controllers/user.ts:38]: okee

**Review by azhariemuhammad — APPROVED** [review]
lgtm

**Review by ibnuimam — APPROVED** [review]

**Pull request merged**